lib/fog/compute/ecloud/models/environments.rb

```diff
@@ -12,8 +12,18 @@ class Environments < Fog::Ecloud::Collection
 
         def all
           data = []
-          service.get_organization(href).body[:Locations][:Location].each do |d|
-            environments = d[:Environments]
+          raw_location = service.get_organization(href).body[:Locations][:Location]
+          if raw_location.is_a?(Array)
+            # If there's more than one location, the XML parser returns an
+            # array.
+            location = raw_location
+          else
+            # Otherwise it returns a hash.
+            location = [raw_location]
+          end
```

Comment:
I believe you could just say `Array(raw_location)`. =)

But this is not a blocker so i will merge it right away!